src/cell_group.hpp

```diff
@@ -277,6 +258,14 @@ class cell_group {
     target_handle get_target_handle(cell_member_type target_id) const {
         return target_handles_[handle_partition_lookup(target_handle_divisions_, target_id)];
     }
+
+    void reset_samplers() {
+        // clear all pending sample events and reset to start at time 0
+        sample_events_.clear();
+        for(uint32_t i=0u; i<samplers_.size(); ++i) {
```

Comment:
While we're here, we should make this `for` consistent/safe (standard integer size vs. `std::size_t` argument.) Happy to do the edit in-place if you've no objection.